scheduler/scheduler.go

```diff
@@ -227,30 +233,117 @@ func (driver *MesosSchedulerDriver) masterDetected(master *mesos.MasterInfo) {
 		}
 
 		driver.MasterPid = pid // save for downstream ops.
+		driver.tryAuthentication()
+	} else {
+		log.Infoln("No master detected.")
+	}
+}
+
+func (driver *MesosSchedulerDriver) tryAuthentication() {
+	if driver.authenticated {
+		// programming error
+		panic("already authenticated")
+	}
+
+	masterPid := driver.MasterPid // save for referencing later in goroutine
+	if masterPid == nil {
+		log.Info("skipping authentication attempt because we lost the master")
+		return
+	}
+
+	if driver.authenticating != nil {
```

Comment:
Using driver.authenticating channel's assigned value (not its content) as a way to signal seems a bit abusive of the channel idiom, no? Are you gaining thread-safety doing it this way ?